app/src/androidTest/java/ch/epfl/skysync/flightdetail/FlightDetailUiTest.kt

```diff
@@ -13,243 +11,168 @@ import androidx.compose.ui.test.onNodeWithText
 import org.junit.Rule
 import org.junit.Test
 
 class FlightDetailUiTest {
   @get:Rule val composeTestRule = createComposeRule()
-  val pilot = Role(RoleType.PILOT, null)
-  val crew = Role(RoleType.CREW, null)
-  val car = Vehicle("Car")
-  lateinit var navController: TestNavHostController
-  var dummyFlight =
-      mutableStateOf(
-          PlannedFlight(
-              UNSET_ID,
-              1,
-              FlightType.FONDUE,
-              Team(listOf()),
-              null,
-              null,
-              LocalDate.now(),
-              TimeSlot.AM,
-              listOf(
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car,
-                  car)))
+
+  @get:Rule val mockkRule = MockKRule(this)
+  // Relaxed mocks methods have a default implementation returning values
+  @RelaxedMockK lateinit var navController: NavHostController
+  val dbs = DatabaseSetup()
+  var flight = dbs.flight1
+  val flightId = flight.id
 
   @Before
-  fun setUpNavHost() {
+  fun setUpNavHost() = runTest {
+    val db = FirestoreDatabase(useEmulator = true)
+    dbs.clearDatabase(db)
+    dbs.fillDatabase(db)
     composeTestRule.setContent {
       FlightDetailUi(
-          backClick = {},
-          deleteClick = {},
-          editClick = {},
-          confirmClick = {},
+          backClick = { navController.popBackStack() },
+          deleteClick = { navController.navigate(Route.HOME) },
+          editClick = { navController.navigate(Route.MODIFY_FLIGHT + "/${flightId}") },
+          confirmClick = { navController.navigate(Route.CONFIRM_FLIGHT + "/${flightId}") },
           padding = PaddingValues(0.dp),
-          flight = dummyFlight.value,
-          flightId = dummyFlight.value.id)
+          flight = flight,
+      )
     }
   }
 
   @Test
-  fun BackButtonIsDisplayed() {
-    composeTestRule.onNodeWithText("Back").assertIsDisplayed()
-  }
-
-  @Test
-  fun BackButtonIsClickable() {
-    composeTestRule.onNodeWithText("Back").assertHasClickAction()
+  fun backButtonWorks() {
+    composeTestRule.onNodeWithText("Back").performClick()
+    verify { navController.popBackStack() }
+    confirmVerified(navController)
   }
 
   @Test
-  fun DeleteButtonIsDisplayed() {
-    composeTestRule.onNodeWithText("Delete").assertIsDisplayed()
+  fun deleteButtonWorksWhenDismiss() {
+    composeTestRule.onNodeWithTag("AlertDialog").assertIsNotDisplayed()
+    composeTestRule.onNodeWithText("Delete").performClick()
+    composeTestRule.onNodeWithTag("AlertDialog").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("AlertDialogDismiss").performClick()
+    composeTestRule.onNodeWithTag("AlertDialog").assertIsNotDisplayed()
```

Comment:
maybe in the end also check that after the dismiss we're still on flight detail page and that no flight has been deleted